environment/core/Halite.cpp

```diff
@@ -719,10 +721,53 @@ auto Halite::process_events() -> void {
             target_count[src] = num_prev_targets + 1;
         };
 
+        auto credit_damage = [&](hlt::EntityId source,
+                                 hlt::EntityId target,
+                                 int damage) -> void {
+            hlt::PlayerId dealer = 0;
+            hlt::PlayerId receiver = 0;
+            switch (source.type) {
+            case hlt::EntityType::PlanetEntity: {
```

Comment:
For the way I think damage should be accounted (each user's ship to anything), just remove this case statement should work.